src/resources/parser/glb-container-resource.js

```diff
@@ -190,6 +192,70 @@ class GlbContainerResource {
         return GlbContainerResource.createSceneHierarchy(sceneClones, 'Entity');
     }
 
+    // get material variants
+    getMaterialVariants() {
+        return Object.keys(this.data.variants);
+    }
+
+    // apply material variant to entity
+    applyMaterialVariant(entity, name) {
+        if (!name) {
+            this._resetMaterialVariant(entity);
+            return;
+        }
+        const variant = this.data.variants[name];
+        if (variant === undefined) {
+            Debug.warn(`No variant named ${name} exists in resource`);
+            return;
+        }
+        const renders = entity.findComponents("render");
+        for (let i = 0; i < renders.length; i++) {
+            const renderComponent = renders[i];
+            this._applyMaterialVariant(variant, renderComponent.meshInstances);
+        }
+    }
+
+    // reset material variants on entity
+    _resetMaterialVariant(entity) {
```

Comment:
I think I'd get rid of these _reset functions, as its the same code as the function above apart from the material selection - just make a small change to that function to handle it and remove the code duplication?